### PR TITLE
feat(input): initial file type implementation

### LINKED ISSUE #6544: IgxInput directive to style file type input
## Is your feature request related to a problem? Please describe.
I am trying to style a file input with `igxInput` directive, but the result is not satisfying.

## Describe the solution you'd like
I would like `igxInput` to be able to style the file input including the button and text.

### PR DESCRIPTION
**Pull request by didimmova**
Closes #6544
Samples - https://github.com/IgniteUI/igniteui-angular-samples/pull/2326
Docs - https://github.com/IgniteUI/igniteui-docfx/pull/2326

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/directives/input/input.directive.ts [hunk 1]
```diff
@@ -220,26 +243,84 @@ export class IgxInputDirective implements AfterViewInit, OnDestroy {
...
+                }
+            }
+
+            this._fileNames = (fileArray || []).map((f: File) => f.name).join(', ');
+
+            if (!this._fileNames) {
+                this._fileNames = this._filePlaceholder;
+            }
+
+            if (this.required && fileList?.length > 0) {
```
Comment (wnvko) [inline projects/igniteui-angular/src/lib/directives/input/input.directive.ts:269]: this should not happen here. Validation should be done on blur

**Review by wnvko — CHANGES_REQUESTED** [review]
Validation of the input is broken now. As input is hidden it does not get focus, neither it can blur.

File: projects/igniteui-angular/src/lib/input-group/input-group.component.html [hunk 2]
```diff
@@ -20,19 +20,48 @@
     <ng-content select="igx-prefix, [igxPrefix]"></ng-content>
 </ng-template>
 
+<ng-template #uploadButton>
```
[the remark this answers is not in the document]
Comment (simeonoff) [reply projects/igniteui-angular/src/lib/input-group/input-group.component.html:23]: Should localization be handled in this PR?

File: projects/igniteui-angular/src/lib/core/i18n/input-resources.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export interface IInputResourceStrings {
+    igx_input_upload_button?: string;
+    igx_input_clear_button?: string;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/i18n/input-resources.ts:3]: You also need a string for the placeholder.

**Review by simeonoff — CHANGES_REQUESTED** [review]

**Comment by kdinev** [comment]
@simeonoff Are you going to finish this PR?

**Comment by simeonoff** [comment]
> @simeonoff Are you going to finish this PR?

@Lipata wanted to take over this PR. We've discussed it several months ago and they wanted to fix some other issues with the input component before incorporating the changes made in this PR. Plus, they didn't really like the implementation introduced here.

**Comment by kdinev** [comment]
@Lipata Is this inactive/abandoned?

**Comment by wnvko** [comment]
With current implementation the validation of the file type input does not work. It is not validating the input in template driven form nor it validate it in template driven forms. The main issue here is that the input is hidden and it never gets focused or blurred, and in the end validation fails.

**Comment by kdinev** [comment]
@wnvko @Lipata Even if validation doesn't work, I would still prefer for the general styling to be merged and the validation to be implemented in the future.

File: projects/igniteui-angular/src/lib/directives/input/input.directive.ts (hunk 1 above)
Comment (wnvko) [reply projects/igniteui-angular/src/lib/directives/input/input.directive.ts:249]: there will be separate PR fixing validation

**Review by wnvko — DISMISSED** [review]

**Review by wnvko — DISMISSED** [review]

**Review by simeonoff — DISMISSED** [review]

**Comment by wnvko** [comment]
In all themes, when you focus the input, there is no visual indication that the input is active.
In Bootstrap themes when you click on `Upload file` button its border goes out of the input.

**Review by simeonoff — DISMISSED** [review]

**Comment by simeonoff** [comment]
> In all themes, when you focus the input, there is no visual indication that the input is active.
> In Bootstrap themes when you click on `Upload file` button its border goes out of the input.
> There is misalignment between button text and input value in material themes
> image

Fixed the first issue related to focusing the input.
The second issue was related to another issue, which was not related to the file type implementation. It was fixed in master and master has been merged into this branch, so all issues should be resolved now.

**Review by wnvko — APPROVED** [review]

**Review by simeonoff — APPROVED** [review]

**Pull request merged**